src/content/docs/logs/ui-data/query-time-parsing.mdx

```diff
@@ -0,0 +1,412 @@
+
+import logsNrqlBehindQueryRule from 'images/logs_screenshot-crop_nrql-behind-query-rule.webp'
+
+import logsDeleteQueryTimeRule from 'images/logs_screenshot-crop_delete-query-time-rule.webp'
+
+import logsViewResultsinLogDetailsView from 'images/logs_screenshot-full_view-results-in-log-details-view.webp'
+
+import logsLogTableResults from 'images/logs_screenshot-full_log-table-results.webp'
+
+import logsRuleManager from 'images/logs_screenshot-crop_rule-manager.webp'
+
+import logsQueryTimeSearchBar from 'images/logs_screenshot-full_query-time-search-bar.webp'
+
+import logsAlertQueryBuilder from 'images/logs_screenshot-crop_alert-query-builder.webp'
+
+import logsQueryBehindParsingRule from 'images/logs_screenshot-crop_query-behind-parsing-rule.webp'
+
+import logsParsingAttributesView from 'images/logs_screenshot-crop_parsing-attributes-view.webp'
+
+
+
+
+
+Query time parsing lets you parse your logs directly in the UI without needing to write complex regular expressions or grok patterns. Results are shown instantly since parsing is performed at query time. You can use query time parsing to temporarily extract values from your logs and quickly perform a query on these variables.
+
+## How query time parsing differs from ingest parsing [#differences]
+
+While both types of parsing make it easier for you to query logs, they have some significant differences:
+
+**Ingest parsing:** Parsing during [log ingestion](/docs/logs/ui-data/parsing/) is where you create parsing rules using grok and/or regex. As log records are ingested at New Relic, your parsing rules are applied to create permanent attributes that are stored with your log data in NRDB. These attributes make it easier for you to query log data. 
+
+**Query time parsing:** In contrast to ingest parsing, query time parsing lets you create temporary attributes that will be used as query variables. You can then use these variables in NRQL queries to populate your log table. We automatically create the queries as you make selections in the UI. 
+
+You may choose to use a combination of both parsing approaches. Review the table below to decide if query time parsing is right for you:
+
+<table>
+    <thead>
+        <tr>
+            <th>
+                Description 
+            </th>
+            <th>
+                Ingest parsing 
+            </th>
+            <th>
+                Query time parsing 
+            </th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                Recommended usage
+            </td>
+            <td>
+                Best for creating permanent attributes you can query in the future
+            </td>
+            <td>
+                Best for doing quick queries on attributes that aren't permanent 
+            </td>
+        </tr>
+        <tr>
+            <td>
+                Parsing language                
+            </td>
+            <td>
+                You create grok patterns and regular expressions 
```

Comment:
Maybe I'm confused actually! What is Grok? Are you speaking about something other than New Relic AI? I've researched it a little and all I am coming up with is Elon Musk's Grok? Is there something I am missing?

Reply:
@homelessbirds I'm going to capitalize Grok since we do that in other docs. It's a special query language for logs that's a dialect of regular expressions.